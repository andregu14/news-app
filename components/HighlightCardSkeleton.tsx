import React from 'react';
import { View } from './Themed';
import { StyleSheet, useColorScheme, useWindowDimensions } from 'react-native';
import Colors from '@/constants/Colors';
import { Skeleton } from './Skeleton';

export default function HighlightCardSkeleton() {
  const colorScheme = useColorScheme() ?? 'light';
  const themeColors = Colors[colorScheme];
  const { width } = useWindowDimensions();
  const cardWidth = Math.min(width * 0.4, 340);

  return (
    <View
      style={[
        styles.container,
        {
          width: cardWidth,
          borderColor: themeColors.borderColor,
          backgroundColor: colorScheme === 'dark' ? '#181A20' : '#FBFBFB',
        },
      ]}
    >
      <Skeleton height={100} width="100%" radius={"square"}/>
      <View style={styles.skeletonTextGroup}>
        <Skeleton height={14} width="90%" />
        <Skeleton height={14} width="70%" />
        <Skeleton height={14} width="80%" />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    height: 200,
    borderRadius: 14,
    marginHorizontal: 7.5,
    borderWidth: 1,
    overflow: 'hidden',
  },
  skeletonTextGroup: {
    padding: 10,
    width: '100%',
    height: '50%',
    justifyContent: 'space-evenly',
  },
});