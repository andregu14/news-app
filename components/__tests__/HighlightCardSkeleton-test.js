import { render, screen } from '@testing-library/react-native';
import HighlightCardSkeleton from '../HighlightCardSkeleton';

// Mock para Skeleton component
jest.mock('../Skeleton', () => ({
    __esModule: true,
    Skeleton: ({ children }) => children
}));

describe('<HighlightCardSkeleton />', () => {
    it('renders correctly', () => {
        render(<HighlightCardSkeleton />);
        expect(screen.toJSON()).toMatchSnapshot();
    });
});
