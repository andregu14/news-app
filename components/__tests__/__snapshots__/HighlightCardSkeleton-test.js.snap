// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HighlightCardSkeleton /> renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      [
        {
          "borderRadius": 14,
          "borderWidth": 1,
          "height": 200,
          "marginHorizontal": 7.5,
          "overflow": "hidden",
        },
        {
          "backgroundColor": "#FBFBFB",
          "borderColor": "#eee",
          "width": 300,
        },
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "height": "50%",
          "justifyContent": "space-evenly",
          "padding": 10,
          "width": "100%",
        },
      ]
    }
  />
</View>
`;
